app/models/signon_identity.rb

```diff
@@ -15,63 +15,66 @@ def from_omniauth(omniauth_auth)
     end
 
     def from_session_data(data)
-      user = User.find(data.fetch('user_id'))
       new(
-        user,
+        User.find(data.fetch('user_id')),
         full_name: data.fetch('full_name'),
         profile_url: data.fetch('profile_url'),
-        logout_url: data.fetch('logout_url')
+        logout_url: data.fetch('logout_url'),
+        available_organisations: data.fetch('available_organisations'),
+        current_organisation: data.fetch('current_organisation')
       )
     rescue KeyError
       raise InvalidSessionData
     end
 
   private
 
-    # rubocop:disable Metrics/MethodLength
     def find_or_create_authorized_user(info)
       email = info.fetch('email')
       permissions = info.fetch('permissions')
 
-      sso_orgs = permissions.map { |p| p.fetch('organisation') }
-      estates = Estate.where(sso_organisation_name: sso_orgs)
```

Comment:
What happens if one of a users permissions does not have a corresponding estate?